pyspeckit/spectrum/models/ammonia.py

```diff
@@ -179,41 +181,43 @@ def ammonia(xarr, tkin=20, tex=None, ntot=14, width=1, xoff_v=0.0,
                                        (Crot-Brot)*Jpara**2)/(kb*tkin))
         Zortho = 2*(2*Jortho+1)*np.exp(-h*(Brot*Jortho*(Jortho+1)+
                                            (Crot-Brot)*Jortho**2)/(kb*tkin))
+        Qpara = Zpara.sum()
+        Qortho = Zortho.sum()
         for linename in line_names:
             if ortho_dict[linename]:
                 orthoparafrac = fortho
                 Z = Zortho
+                Qtot = Qortho
                 count = ortho_count
                 ortho_count += 1
             else:
                 orthoparafrac = 1.0-fortho
                 Z = Zpara
+                Qtot = Qpara
                 count = para_count # need to treat partition function separately
                 para_count += 1
 
+
+            # for a complete discussion of these equations, please see
+            # https://github.com/keflavich/pyspeckit/blob/ammonia_equations/examples/AmmoniaLevelPopulation.ipynb
+            # and
+            # http://low-sky.github.io/ammoniacolumn/
+
             # short variable names for readability
             frq = freq_dict[linename]
             partition = Z[count]
             aval = aval_dict[linename]
 
-            # Friesen 2009 eqn A4 points out that the partition function actually says
-            # how many molecules are in the NH3(1-1) state, both upper *and* lower.
-            # population_upperlower = ntot * orthoparafrac * partition/(Z.sum())
-            # population_upperstate = population_upperlower / (1+np.exp(h*frq/(kb*tex)))
-            #
-            # Note Jan 1, 2015: This is accounted for in the eqn below.  The
-            # only difference is that I have used Tkin where Friesen et al 2009
-            # use Tex.  Since Tex describes which states are populated, that may
-            # be the correct one to use.
-
             # Total population of the higher energy inversion transition
-            population_upperstate = lin_ntot * orthoparafrac * partition/(Z.sum())
+            population_upperstate = lin_ntot * orthoparafrac * partition/Qtot
```

Comment:
Agreed that simplifications and be obfuscating.  For example, we should probably be taking `g_(1,1)_a = 5` and `g_(1,1)_s = 5` as per Figure 12 of Mangum and Shirley.  At least they are the same number of internal states.  I'm still confused about C17O.  I'm also confused as to why that Figure labels the - parity (i.e., antisymmetric) as the lower energy state.

All right: we have a clear difference of interpretation.  I would contend that `N(J,J)_s` and `N(J,J))_a` are connected by `Tex` and not `Tkin` using this as the definition of `Tex`, namely the temperature you plug into Boltzmann to get the ratio between the upper and lower levels of a line-forming pair of states.